app/services/checkout/form_data_adapter.rb

```diff
@@ -31,6 +33,18 @@ def move_payment_source_to_payment_attributes!
       @params[:order][:payments_attributes].first[:source_attributes] = payment_source_params
     end
 
+    def fill_in_card_type
+      payment = params[:order][:payments_attributes]&.first&.dig(:source_attributes)
+
+      return if payment&.dig(:number).blank?
```

Comment:
if payment is nil this will not return. I think it should be
return unless payment&.dig(:number).present?

Reply:
I think this is how I'd write this method:

```ruby
def fill_in_card_type
     return unless payment_source_attributes
     
     return if payment_source_attributes.dig(:number).blank?

     payment[:cc_type] ||= card_brand(payment_source_attributes[:number])
end

def payment_source_attributes
    @payment_source_attributes ||= params[:order][:payments_attributes]&.first&.dig(:source_attributes)
end
```

Reply:
> if payment is nil this will not return.

Really? I don't understand. `nil&....` returns nil and `nil.blank?` is true.

```rb
nil&.dig(:number).blank?
=> true
```

Reply:
:-( you are right, I was not aware that the chain would be followed if the conditional navigator & returned nil.
I am sorry about that.
so this method could be:
```
def fill_in_card_type    
     return if payment_source_attributes&.dig(:number).blank?

     payment[:cc_type] ||= card_brand(payment_source_attributes[:number])
end
```
I dont think it's a major difference, otherwise I'd open a PR.

I am sorry I moved this forward and the code is not as you wanted. I did it to get this merged asap. Sorry Maikel.

Reply:
No, no, I'm so happy you took over! We needed to solve this quickly. This bit of code style is not that important compared to the bug we are solving.

Reply:
:-)